cmake/config_validation.cmake

```
@@ -1,71 +1,138 @@
-# Name: config_validation.cmake file
-# ThundeRatz Robotics Team
-# Brief: This file contains the existence checks of the declared variables
-# 04/2023
-
 ###############################################################################
-## Auxiliary Sets
+## Existence checks
 ###############################################################################
 
-# This variable is used by the stm32-cmake lib to find the STM32CubeMX files
-# @see: https://github.com/ObKo/stm32-cmake#configuration
-set(STM32_CUBE_${DEVICE_CORTEX}_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cube)
-
-# This set contains all the variables that must be defined by the user
-# It is used to check if all of them are properly defined
-set(USER_INPUT_VARIABLES
-    DEVICE_CORTEX
-    DEVICE_FAMILY
-    DEVICE_TYPE
-    DEVICE_DEF
-    DEVICE_FAMILY_COMPACT
-    DEVICE
-    BOARD_VERSION
-    TARGET_BOARD
-)
+# Check if CMake build type is correctly configured
+if(NOT (BUILD_TYPE STREQUAL "Release"        OR BUILD_TYPE STREQUAL "Debug" OR
+        BUILD_TYPE STREQUAL "RelWithDebInfo" OR BUILD_TYPE STREQUAL "MinSizeRel"))
+    set(BUILD_TYPE "RelWithDebInfo")
+endif()
 
-###############################################################################
-## Existence checks
-###############################################################################
+set(CMAKE_BUILD_TYPE ${BUILD_TYPE})
+message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
 
-## Check if STM32CubeMX variables are properly defined
-if(DEFINED ENV{CUBE_PATH})
-    message(STATUS "CUBE_PATH defined as $ENV{CUBE_PATH}")
-else()
-    message(FATAL_ERROR "CUBE_PATH not defined")
+# Check if host system is Linux
+if(NOT CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+    message(FATAL_ERROR "${CMAKE_HOST_SYSTEM_NAME} is not supported")
 endif()
 
-if(CMAKE_HOST_WIN32)
-    set(JAVA_EXE "$ENV{CUBE_PATH}\\STM32CubeMX.exe")
-    set(CUBE_JAR "$ENV{CUBE_PATH}\\jre\\bin\\java.exe")
-    set(JLINK_EXE JLink.exe)
+# Detect if running inside WSL
+if(DEFINED ENV{WSL_DISTRO_NAME})
+    message(STATUS "WSL detected")
+    set(CUBE_DEFAULT_PATH "/mnt/c/Program Files/STMicroelectronics/STM32Cube/STM32CubeMX")
+    set(CUBE_PROGRAM "STM32CubeMX.exe")
+    set(PROGRAMMER_CMD "STM32_Programmer_CLI.exe")
+    set(JLINK_CMD "JLink.exe")
+    set(OPENOCD_CMD "openocd.exe")
 else()
-    set(JAVA_EXE $ENV{CUBE_PATH}/jre/bin/java)
-    set(CUBE_JAR $ENV{CUBE_PATH}/STM32CubeMX)
-    set(JLINK_EXE JLinkExe)
+    message(STATUS "Linux detected")
+    set(CUBE_DEFAULT_PATH "/usr/local/STMicroelectronics/STM32Cube/STM32CubeMX")
```

```suggestion
    if (CMAKE_HOST_APPLE)
        message(STATUS "MacOS detected")
        set(CUBE_DEFAULT_PATH "/Applications/STMicroelectronics/STM32CubeMX.app")
    else()
        message(STATUS "Linux detected")
        set(CUBE_DEFAULT_PATH "/usr/local/STMicroelectronics/STM32Cube/STM32CubeMX")
    endif()

```

Tem que pedir pra alguém com Mac testar se só isso é suficiente.